01-Login/README.md

```diff
@@ -1,84 +1,705 @@
+        if (err) {
+          reject(err);
+        } else {
+          this.localLogin(authResult);
+          resolve(authResult.idToken);
+        }
+      });
+    });
+  }
+
+  localLogin(authResult) {
+    this.idToken = authResult.idToken;
+    this.profile = authResult.idTokenPayload;
+
+    // Convert the JWT expiry time from seconds to milliseconds
+    this.tokenExpiry = new Date(this.profile.exp * 1000);
+
+    localStorage.setItem(localStorageKey, 'true');
+
+    this.emit(loginEvent, {
+      loggedIn: true,
+      profile: authResult.idTokenPayload,
+      state: authResult.appState || {}
+    });
+  }
+
+  renewTokens() {
+    return new Promise((resolve, reject) => {
+      if (localStorage.getItem(localStorageKey) !== "true") {
+        return reject("Not logged in");
+      }
+      
+      webAuth.checkSession({}, (err, authResult) => {
+        if (err) {
+          reject(err);
+        } else {
+          this.localLogin(authResult);
+          resolve(authResult);
+        }
+      });
+    });
+  }
+
+  logOut() {
+    localStorage.removeItem(localStorageKey);
+
+    this.idToken = null;
+    this.tokenExpiry = null;
+    this.profile = null;
+
+    webAuth.logout({
+      returnTo: window.location.origin
+    });
+
+    this.emit(loginEvent, { loggedIn: false });
+  }
+
+  isAuthenticated() {
+    return (
+      Date.now() < this.tokenExpiry &&
+      localStorage.getItem(localStorageKey) === 'true'
+    );
+  }
+}
+
+export default new AuthService();
 ```
 
-## Deployment
+The service now includes several other methods for handling authentication.
 
-### Compiles and minifies for production
+- `handleCallback` - looks for an authentication result in the URL hash and processes it with the `parseHash` method from auth0.js
+- `localLogin` - sets the user's ID Token, and a time at which the ID Token will expire. The expiry time is converted to milliseconds so that the native JavaScript `Date` object can be used
+- `renewTokens` - uses the `checkSession` method from auth0.js to renew the user's authentication status, and calls `localLogin` if the login session is still valid
+- `logout` - removes the user's tokens from memory. It also calls `webAuth.logout` to log the user out at the authorization server
+- `isAuthenticated` - checks whether the local storage flag is present and equals "true", and that the expiry time for the ID Token has passed
 
-```bash
-npm run build
+### About the Authentication Service
+
+When you set up the `AuthService` service, you create an instance of the `auth0.WebAuth` object. In that instance, you can define the following:
+
+* Configuration for your application and domain
+* Response type, to show that you need a user's ID Token after authentication
+* The URL where you want to redirect your users after authentication.
+
+::: note
+In this tutorial, the route is `/callback`, which is implemented in the [Add a Callback Component](#add-a-callback-component) step. 
```

Comment:
Can you check that these anchor links work when the readme is rendered out in HTML?